### PR TITLE
Trademarks refactor

### PR DESCRIPTION
**Pull request by hschoenburg**
This is ready to go.

There is one name, "spontaneous" the owner of which did everything we asked but did not provide any easily identifiable domain for us to attribute to. Waiting for them to reply via email but as a stop-gap, assigned the name to handshake.org, and we can transfer it to them post launch if needed.  

Will update this PR with a domain for "spontaneous" as soon as I receive one.

### CODE REVIEW

File: names/trademarks.json [hunk 1]
```diff
@@ -1,166 +1,298 @@
...
+    "comunitaria",
+    "comunitaria.com"
+  ],
+  [
+    "iocom",
+    "iocom.com"
+  ],
+  [
+    "spontaneous",
+    "handshake.org"
```
Comment (hschoenburg) [inline names/trademarks.json:64]: using handshake.org here as placeholder. Waiting for trademark owner to provide domain

Comment (chjj) [reply names/trademarks.json:64]: Any update on this?

**Comment by chjj** [comment]
It looks like these ones got renamed:

- `origin` -> `originalprotocol` (they actually had both before, but `origin` got removed)
- `notation` -> `notationcapital`
- `explore` -> `useexplore`

Should we keep both names for each? Above commit has a diff that illustrates this better.

edit: Also looks like a bunch got added (ones that weren't in the address file before).

**Comment by chjj** [comment]
Also, why is `metamask` set to `gmail.com`?

**Comment by chjj** [comment]
I did some more analysis. A significant majority of these names will already get them through the alexa reserved list, so we can probably just remove them from the trademark list. A few other names will need to be "aliased" on the consensus layer since their lowest subdomain doesn't match the domain they want (for example, we can alias `londontrustmedia.com` to `privateinternetaccess`). The creates a problem: what if they have two names for the same domain? Or what if they already have their domain in the alexa top 100k but want a _different_ name?

My analysis shows that we can handle all of these with the regular name proofs with minimal changes (aliases), except for: `userland` (collision with privateinternetaccess), `tikiwiki` (collision with tiki), and `metamask` (only because right now it's set to gmail for some reason -- can be fixed).

My script's output:

```
Warning: londontrustmedia.com will not get their name of userland because they have duplicate names which require an alias!
Warning: gmail.com will not get their name of metamask because they already own gmail!
Warning: tiki.org will not get their name of tikiwiki because they already own tiki!
Note: a16z.com already owns a16z!
Note: binance.com already owns binance!
Note: blender.org already owns blender!
Note: blockfolio.com already owns blockfolio!
Note: buddy.works already owns buddy!
Note: cakephp.org already owns cakephp!
Note: codewars.com already owns codewars!
Note: coingecko.com already owns coingecko!
Note: contrib.com already owns contrib!
Note: district0x.io already owns district0x!
Note: freenode.net already owns freenode!
Note: gainesvillecoins.com already owns gainesvillecoins!
Note: gnome.org already owns gnome!
Note: infura.io already owns infura!
Note: jaxx.io already owns jaxx!
Note: lbry.io already owns lbry!
Note: me3d.com.au already owns me3d!
Note: mixbook.com already owns mixbook!
Note: namecheap.com already owns namecheap!
Note: nextdoor.com already owns nextdoor!
Note: paddle8.com already owns paddle8!
Note: pewresearch.org already owns pewresearch!
Note: showclix.com already owns showclix!
Note: tierion.com already owns tierion!
Note: tiki.org already owns tiki!
Note: wikihow.com already owns wikihow!

Necessary aliases:
astraldynamics.co.uk -> astralship
gov.bm -> bermuda
cryptionary.io -> cryptopedia
primekey.se -> ejbca
tokensoft.io -> knoxwallet
glop.org -> mackup
getpaloma.com -> paloma
londontrustmedia.com -> privateinternetaccess
carlson.net -> socialchess
handshake.org -> spontaneous
vaporwa.re -> vaporware

{ names:
   [ [ 'a16zcrypto',
       'a16zcrypto.com',
       'hs1q86q9ska4j263p3cmgycqfrxgnjhrvs227uzpz2' ],
     [ 'agari', 'agari.com', null ],
     [ 'anchorage',
       'anchorage.com',
       'hs1qtrnme028c9q73avehqs9ujrygle7tdja6wuz97' ],
     [ 'angellikefire', 'angellikefire.com', null ],
     [ 'barcraft', 'barcraft.com', null ],
     [ 'base58',
       'base58.capital',
       'hs1qmgkgm83zx6mydkec0s5ml0vx5yur8c59hejew6' ],
     [ 'bubblestudent', 'bubblestudent.co.uk', null ],
     [ 'candybar',
       'candybar.co',
       'hs1q53vx5qplut3422evphha5pvu6j0rmjr4eqrse3' ],
     [ 'coindera',
       'coindera.com',
       'hs1q7q3h9yrv0qj3s34a8mw7j55krquulhcxvvgd2s' ],
     [ 'comunitaria',
       'comunitaria.com',
       'hs1qd2u8anwt893pduenjfckhdgeataet37j362vkf' ],
     [ 'decentral', 'decentral.ca', null ],
     [ 'documize', 'documize.com', null ],
     [ 'ecorp', 'ecorp.com', null ],
     [ 'fossa', 'fossa.io', null ],
     [ 'grownome', 'grownome.com', null ],
     [ 'iocom',
       'iocom.com',
       'hs1qd2u8anwt893pduenjfckhdgeataet37j362vkf' ],
     [ 'longgame', 'longgame.co', null ],
     [ 'mattslater',
       'mattslater.co',
       'hs1q70tg0mltmxpxk7wmhqq3czw3h5entkya2zqsef' ],
     [ 'maxsys', 'maxsys.ai', null ],
     [ 'nettalk', 'nettalk.com', null ],
     [ 'nexves', 'nexves.com', null ],
     [ 'notationcapital',
       'notationcapital.com',
       'hs1q0v0084ztp484f6mh5u9sadl73vm30ntwd6pjna' ],
     [ 'num', 'num.uk', 'hs1q3lj294eg2xvax3k7jk8z66u8tfpqhuwek064qc' ],
     [ 'nxlog',
       'nxlog.org',
       'hs1qsh6gwrvr52e2xggzekr0t0at6clly8xhur3fyf' ],
     [ 'organism',
       'organism.ai',
       'hs1qu96rw66rqhntw9nxhmtest39frl44vjpq7cplc' ],
     [ 'originprotocol',
       'originprotocol.com',
       'hs1qnm7gfurl2duexukk8dlnseuzpgpgjf9t54ugc4' ],
     [ 'snapship', 'snapship.it', null ],
     [ 'tech2mkt', 'tech2mkt.com', null ],
     [ 'tenonedesign',
       'tenonedesign.com',
       'hs1qp4sz5rvarq29ushgz77kh4yslgh80u9r7ntgmx' ],
     [ 'tokensoft', 'tokensoft.io', null ],
     [ 'tripletiedout', 'tripletiedout.com', null ],
     [ 'unitychain', 'unitychain.net', null ],
     [ 'useexplore',
       'useexplore.com',
       'hs1qgwjpzc6hwhs6enp2em99nlj262dqwmaztpj8u5' ],
     [ 'wolk',
       'wolk.com',
       'hs1qr3vfmaf89fsluql6pssdzjxx78deujdfgkyv78' ] ],
  alias:
   [ [ 'astralship', 'astraldynamics.co.uk', null ],
     [ 'bermuda',
       'gov.bm',
       'hs1q0ed63gs4ecaqyderxy4jh3eyxld39cuh0aw8hd' ],
     [ 'cryptopedia', 'cryptionary.io', null ],
     [ 'ejbca',
       'primekey.se',
       'hs1qlmsjqxsgkn5almuuygwqyxsrmgf53wyv65npr4' ],
     [ 'knoxwallet', 'tokensoft.io', null ],
     [ 'mackup',
       'glop.org',
       'hs1qzejy7lu83uvkplyqc26z0weneq92zww99zcszs' ],
     [ 'paloma', 'getpaloma.com', null ],
     [ 'privateinternetaccess',
       'londontrustmedia.com',
       'hs1qy4r9ew2e3yaa2g7dhepfsn4vsjydszgtl2cfgp' ],
     [ 'socialchess', 'carlson.net', null ],
     [ 'spontaneous',
       'handshake.org',
       'hs1qe87af6nq4ctwf4h60ell4mqqnc5tyetaslx5vl' ],
     [ 'vaporware', 'vaporwa.re', null ] ],
  dupsAlias:
   [ [ 'userland',
       'londontrustmedia.com',
       'hs1qrqchz5xdzmql8dddz504cy6kage9e9tdjzjgtu' ] ],
  owned:
   [ [ 'a16z',
       'a16z.com',
       'hs1q86q9ska4j263p3cmgycqfrxgnjhrvs227uzpz2' ],
     [ 'binance',
       'binance.com',
       'hs1qprh4ncdjlx6dxdxwquzn5pcwt444epujulfpxf' ],
     [ 'blender',
       'blender.org',
       'hs1qutqe6mhwsnjzrvs9gl5mr258azsvywshn5lapf' ],
     [ 'blockfolio',
       'blockfolio.com',
       'hs1qvw2khen2vdjmgfj5e3pycknxpqt7aerajs09l5' ],
     [ 'buddy',
       'buddy.works',
       'hs1qyfla867rn3rfm2pekq55a7y7zfyaqltl6emfjm' ],
     [ 'cakephp', 'cakephp.org', null ],
     [ 'codewars', 'codewars.com', null ],
     [ 'coingecko', 'coingecko.com', null ],
     [ 'contrib', 'contrib.com', null ],
     [ 'district0x',
       'district0x.io',
       'hs1qfeg0ej42apnuapcsj85l3el78nvs5a87xz44l3' ],
     [ 'freenode',
       'freenode.net',
       'hs1qnvkypvp5uchnujj8awl8th0lm998m0c7c3pf5y' ],
     [ 'gainesvillecoins', 'gainesvillecoins.com', null ],
     [ 'gnome',
       'gnome.org',
       'hs1qcmglasetrultpqcukkxjkgeuf0kdjrn0fc69xk' ],
     [ 'infura',
       'infura.io',
       'hs1q929z3fp79k9zgt95k3cfvsfh4xkjlmj06qsr4y' ],
     [ 'jaxx', 'jaxx.io', null ],
     [ 'lbry',
       'lbry.io',
       'hs1qtku7qtwutu72ukgdngx8u473a5fjcjh4lfqkhr' ],
     [ 'me3d', 'me3d.com.au', null ],
     [ 'mixbook',
       'mixbook.com',
       'hs1qga5450c5kkj5wm6jhdq2cq0ztv6uarz2rmuh8r' ],
     [ 'namecheap',
       'namecheap.com',
       'hs1qy84har66qxjnd27e8c5y2tkwsdhltcmpm8nzdu' ],
     [ 'nextdoor', 'nextdoor.com', null ],
     [ 'paddle8', 'paddle8.com', null ],
     [ 'pewresearch', 'pewresearch.org', null ],
     [ 'showclix',
       'showclix.com',
       'hs1qp05mlr3sm3m7wvw7qkv3t980ttukhlt2f6c8m0' ],
     [ 'tierion',
       'tierion.com',
       'hs1qm5u7eurf2tc7kff6rj5uyjj4cwptzpylg4jfp9' ],
     [ 'tiki',
       'tiki.org',
       'hs1qqyjf3r3rd3z49nrm525yukyuspq60xk0eam8du' ],
     [ 'wikihow',
       'wikihow.com',
       'hs1qa5k70ggnwq63j3qr08du5fahmkyf8plyxuq3lx' ] ],
  dupsOwned:
   [ [ 'metamask',
       'gmail.com',
       'hs1qq5q6g09hnk3cn4ywr9kn2alcxt8aqjf9njnn2d' ],
     [ 'tikiwiki',
       'tiki.org',
       'hs1qqyjf3r3rd3z49nrm525yukyuspq60xk0eam8du' ] ] }
```

**Comment by chjj** [comment]
I've done two things now to resolve some of these conflicts:

1. Replaced `['privateinternetaccess', 'londontrustmedia.com']` with `['privateinternetaccess', 'privateinternetaccess.com']`.
2. Replaced `['metamask', 'gmail.com']` with `['metamask', 'metamask.io']`.

Might want to check with them to make sure that's right.

The only name that's totally unresolvable is `['tikiwiki', 'tiki.org']` since they're already getting `['tiki', 'tiki.org']`. Might want to check with them to see if they have another domain available.

File: names/trademarks.json (hunk 1 above)
Comment (hschoenburg) [reply names/trademarks.json:64]: They haven't replied but doing a little internet stalking they clearly are a big FOSS contributor and control niftylettuce.com.

**Comment by hschoenburg** [comment]
Alright, thanks for the thorough review and my apologies for the dumb mistakes. (sorry metamask and origin, I promise it wasn't personal).

For `spontaneous` we can use `niftylettuce.com`

`notation`, `explore` and `origin` were requested but the requesters don't hold those actual trademarks (they hold 'notationcapital', 'useexplore' and 'originprotocol')

I've asked @marclaporte at TikiWiki to ask for an additional domain, perhaps 'wikisuite.org'.

Also emailed @dpazdan at Metamask about using 'metamask.io'

Will update this PR as responses come in
https://github.com/handshake-org/hs-names/pull/2

**Comment by chjj** [comment]
I made a mistake in my original analysis. `['knoxwallet', 'tokensoft.io']` is also unresolvable, since tokensoft.io is already getting `tokensoft`.

`['spontaneous', 'niftylettuce.com']` means that they won't get `niftylettuce.com`.

I think I have a solution for this. We can tie the colliding names to subdomains: e.g. `spontaneous` goes to `spontaneous.niftylettuce.com`. `knoxwallet` goes to `knoxwallet.tokensoft.io` and so on. The claimants will just have to create a subdomain with a new signed zone. If they actually own their keys, they don't even need to "create" a new subdomain. Just add it to the zone file for a second and run dnssec-signzone.

So, just to be clear, `tikiwiki` should be fine now. They can redeem it with `tikiwiki.tiki.org`.

**Comment by chjj** [comment]
Alright, new list is built. I think this is a pretty elegant solution and fixes all our problems.

**Comment by hschoenburg** [comment]
awesome!

**Comment by pinheadmz** [comment]
I'm trying to understand how the trademark names are integrated... are they all claimed with DNSSEC proofs just like any other reserved name? Meaning, the faucet-like addresses provided by trademark name owners are NOT used anymore?

**Pull request merged**